operationsgateway_api/src/records/export_handler.py

```diff
@@ -0,0 +1,428 @@
+        if len(self.projection) == 1:
+            channel_name = self.projection[0].split(".")[1]
+            filename += "_" + channel_name
+        return filename
+
+    @staticmethod
+    def get_first_last_record_ids(record_ids: List) -> Tuple[str, str]:
+        """
+        Get the first and last record IDs by ordering the list of record IDs and then
+        returning the first and last items.
+        """
+        record_ids_sorted = sorted(record_ids)
+        first = record_ids_sorted[0]
+        if len(record_ids_sorted) == 1:
+            last = None
+        else:
+            last = record_ids_sorted[-1]
+        return first, last
+
+    @staticmethod
+    def add_value_to_csv_line(line, value) -> str:
+        """
+        Helper function for writing values to CSV files
+        """
+        if value is None or value == "":
+            # leave cell empty in these cases
+            return line + ","
+        if type(value) == str:
+            # put quotes round string values in case they contain a comma
+            # which would upset the formatting
+            return line + '"' + value + '"' + ","
+        else:
+            # just put the raw value into the CSV and
+            # Excel should do its best to interpret the type
+            return line + str(value) + ","
+
+    @staticmethod
+    def get_value_from_dict(record_dict, projection) -> Union[str, int, float, Dict]:
+        """
+        Recursive function to extract a value from a record dictionary by traversing
+        down the dictionary following a path specified by the projection to reach a
+        scalar value which is then returned.
+        The recursion handles values that are found at different depths in the record
+        dictionary such as metadata.shotnum and channels.N_COMP_FF_E.data.
+        """
+        # get the projection part up to the first dot
+        projection_array = projection.split(".")
+        next_projection_part = projection_array[0]
+        try:
+            # get either the entry in the dictionary
+            # or a value if we have reached the value we need
+            dict_or_value = record_dict.get(next_projection_part)
+        except AttributeError:
+            # there is no value for this item in this record - this is probably OK
+            # return empty string so that alignment is maintained in the CSV file
+            return ""
+        if len(projection_array) == 1:
+            # we should have reached an item value in the most nested dictionary
+            # so return it
+            return dict_or_value
+        else:
+            # we still have a dictionary which needs further processing
+            # to get down to lower levels where the value will be
+            new_projection = projection.split(".", 1)[1]
+            return ExportHandler.get_value_from_dict(dict_or_value, new_projection)
+
+    @staticmethod
+    def check_zip_file_size(zip_file_in_memory: io.BytesIO) -> None:
```

Comment:
All the code that calls this function are within this class and none of the functions where the calls occur are static methods. The value that's passed in is always part of `self` too. Does this function need to be static?